.github/workflows/update_flake_lock.yml

```diff
@@ -0,0 +1,21 @@
+name: Update Flake Lock File
```

Comment:
**thought:** Not sure if auto-updating flake lock is a good idea. We don't do it in Fedimint, and just use the `self-care` check to remind us to update it every 30 days manually. 

Generally changes to the lock file often result in developers needing to download bunch of data (gigabytes sometimes if cross-compilation is involved). And there's always a chance that something breaks and wasn't caught by the CI, so it's easier to figure it out if tiny lock file changes are not happening too often.

Reply:
Since this is a PR, it will run all tests on CI to make sure that the `flake.lock` update won't rugpull us.